src/screens/PinScreen.tsx

```diff
@@ -0,0 +1,328 @@
+      }}
+    >
+      <TouchableOpacity
+        style={{
+          width: 38,
+          height: 38,
+          justifyContent: 'center',
+          alignItems: 'center',
+        }}
+        onPress={(): void => {
+          props.onPress && props.onPress()
+        }}
+      >
+        <Ionicons name="close" size={28} color={COLOR.black._900} />
+      </TouchableOpacity>
+    </View>
+  )
+}
+
+const PinTitle = (props: { title: string }): ReactElement => {
+  return (
+    <FormText fontType="B.20" color={COLOR.black._900} style={{ margin: 12 }}>
+      {props.title}
+    </FormText>
+  )
+}
+
+const PinSubTitle = (props: {
+  icon?: string
+  title?: string
+  onPress?: () => Promise<void>
+}): ReactElement => {
+  return (
+    <TouchableOpacity
+      style={{ flexDirection: 'row', alignItems: 'center' }}
+      onPress={props.onPress}
+    >
+      {props.icon && (
+        <Ionicons
+          name={props.icon}
+          size={16}
+          color={COLOR.black._500}
+          style={{ marginRight: 4 }}
+        />
+      )}
+      <FormText fontType="R.14" color={COLOR.black._500}>
+        {props.title}
+      </FormText>
+    </TouchableOpacity>
+  )
+}
+
+const PinDot = (props: { fill: boolean }): ReactElement => {
+  const backgroundColor = props.fill ? COLOR.primary._400 : COLOR.black._100
+
+  return (
+    <View
+      style={{
+        width: 16,
+        height: 16,
+        backgroundColor,
+        borderRadius: 16 / 2,
+        margin: 14,
+      }}
+    />
+  )
+}
+
+const PinButton = (
+  props:
+    | { value: string; handleInput: (value: string) => void }
+    | { delete: boolean; handleDelete: () => void }
+    | { disabled: boolean }
+): ReactElement => {
+  const [backgroundColor, setBackgroundColor] = useState<string | undefined>(
+    undefined
+  )
+
+  const onPressIn = (): void => {
+    setBackgroundColor(COLOR.black._90005)
+
+    if ('value' in props) {
+      props.handleInput(props.value)
+    }
+    if ('delete' in props) {
+      props.handleDelete()
+    }
+  }
+  const onPressOut = (): void => {
+    setTimeout(() => setBackgroundColor(undefined), 50)
+  }
+
+  return (
+    <TouchableWithoutFeedback
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
+      disabled={props.hasOwnProperty('disabled')}
+    >
+      <View
+        style={{
+          width: 84,
+          height: 64,
+          borderRadius: 10,
+          backgroundColor,
+          alignItems: 'center',
+          justifyContent: 'center',
+          marginVertical: 12,
+          marginHorizontal: 8,
+        }}
+      >
+        {'delete' in props && (
+          <Image source={images.pin_delete} style={{ width: 32, height: 32 }} />
+        )}
+        {'value' in props && (
+          <FormText fontType="B.32" color={COLOR.black._900}>
+            {props.value}
+          </FormText>
+        )}
+      </View>
+    </TouchableWithoutFeedback>
+  )
+}
+
+const PIN_COUNT = 4
+
+const PinScreen = (): ReactElement => {
```

Comment:
is the design/UX to use the bottomsheetdialog? like with slide from bottom animation

Reply:
PinScreen is used a navigation screen.